lib/core/elastic_connector_actions.rb

```diff
@@ -114,11 +114,24 @@ def claim_job(connector_id)
           :connector_id => connector_id,
           :status => Connectors::SyncStatus::IN_PROGRESS,
           :worker_hostname => Socket.gethostname,
-          :created_at => Time.now
+          :created_at => Time.now,
+          :filtering => convert_connector_filtering_to_job_filtering(connector_record.dig('_source', 'filtering'))
         }
-        job = client.index(:index => Utility::Constants::JOB_INDEX, :body => body)
 
-        job['_id']
+        client.index(:index => Utility::Constants::JOB_INDEX, :body => body)
+      end
+
+      def convert_connector_filtering_to_job_filtering(connector_filtering)
+        return [] unless connector_filtering
+        connector_filtering = [connector_filtering] unless connector_filtering.is_a?(Array)
+        connector_filtering.each_with_object([]) do |filtering_domain, job_filtering|
+          job_filtering << {
+            'domain' => filtering_domain['domain'],
```

Comment:
Why is `domain` not in `active.rules`?

Reply:
From what I remember `active` is the current rules - to not mix up rules in progress of editing and currently used rules?

Reply:
Actually even more specific question - what is `domain`?

Reply:
See the "Notes" section in the [`In .elastic-connectors`](https://docs.google.com/document/d/1llPDGzduj8lST0N45v_G4PB0s0RUOBFurOIsBAD-Fyg/edit?pli=1#heading=h.l2heyce8rz9r). The `domain` was originally added with Crawler in mind. Crawler has different sets of crawl rules for each domain, so we needed to make a list/array structure rather than just an Object under `filtering`. I kept the key name as `domain` because it's more general meaning (like in DDD or DSL) works here - we want to define the "domain" in which these rules apply.

I see a not-too-distant future where we have several domains for MySQL so that we can define different configs and filtering rules for different databases/tables. Would also make it easier for us to support multiple Mongo collections in a single connector in the future.